pkg/kp/constants.go

```diff
@@ -1,38 +1,84 @@
 package kp
 
 import (
-	"strings"
+	"path"
+
+	"github.com/square/p2/pkg/types"
+	"github.com/square/p2/pkg/util"
 )
 
+// PodPrefix represents the top level of a subtree whose leaves are expected to
+// be pods, e.g. "hooks", "intent", "reality"
+type PodPrefix string
+
 const (
-	INTENT_TREE  string = "intent"
-	REALITY_TREE string = "reality"
-	HOOK_TREE    string = "hooks"
-	LOCK_TREE    string = "lock"
-	RC_TREE      string = "replication_controllers"
-	ROLL_TREE    string = "rolls"
+	INTENT_TREE  PodPrefix = "intent"
+	REALITY_TREE PodPrefix = "reality"
+	HOOK_TREE    PodPrefix = "hooks"
+	LOCK_TREE    string    = "lock"
 )
 
-func IntentPath(args ...string) string {
-	return strings.Join(append([]string{INTENT_TREE}, args...), "/")
-}
+var (
+	VALID_POD_PREFIXES = []PodPrefix{
+		INTENT_TREE,
+		REALITY_TREE,
+		HOOK_TREE,
+	}
+)
 
-func RealityPath(args ...string) string {
-	return strings.Join(append([]string{REALITY_TREE}, args...), "/")
-}
+func nodePath(podPrefix PodPrefix, nodeName string) (string, error) {
+	validPrefix := false
+	for _, prefix := range VALID_POD_PREFIXES {
```

Comment:
You don't need `VALID_POD_PREFIXES` anymore. The type system should catch the mistakes.